javalib/src/main/scala/java/security/cert/CertPath.scala

```diff
@@ -1,7 +1,16 @@
 package java.security.cert
 
-import java.util.List
-
 abstract class CertPath protected (`type`: String) {
-  def getCertificates(): List[_ <: Certificate]
+
+  def getCertificates(): java.util.List[_ <: Certificate]
+
+  def getType(): String = `type`
+
+  override def equals(other: Any): Boolean =
+    other match {
+      case _ if this == other => true
```

Comment:
Don't you mean `this eq other` here?

Reply:
You are right, but this will only work if:

```scala
override def equals(other: Any): Boolean =
    other match {
      case otherRef : AnyRef if this eq otherRef => true
      case otherCp: CertPath if otherCp.getType.equals(`type`) =>
        getCertificates().equals(otherCp.getCertificates)
      case _ => false
    }
```